align/main.py

```diff
@@ -293,33 +297,39 @@ def schematic2layout(netlist_dir, pdk_dir, netlist_file=None, subckt=None, worki
         logger.info(f"READ file: {netlist} subckt={subckt}, flat={flatten}")
 
         topology_dir.mkdir(exist_ok=True)
-        primitives = generate_hierarchy(netlist, subckt, topology_dir, flatten, pdk_dir, uniform_height)
+        primitives, ckt_data = generate_hierarchy(netlist, subckt, topology_dir, flatten, pdk_dir, uniform_height)
 
-        gen_more_primitives( primitives, topology_dir, subckt)
+        gen_more_primitives(primitives, topology_dir, subckt)
 
-        with (topology_dir / '__primitives__.json').open( 'wt') as fp:
-            json.dump( primitives, fp=fp, indent=2)
+        with (topology_dir / '__primitives__.json').open('wt') as fp:
+            json.dump(primitives, fp=fp, indent=2)
     else:
         if subckt is None:
-            subckt = extract_netlist_files(netlist_dir,netlist_file).stem
+            subckt = extract_netlist_files(netlist_dir, netlist_file).stem
 
-        with (topology_dir / '__primitives__.json').open( 'rt') as fp:
+        with (topology_dir / '__primitives__.json').open('rt') as fp:
             primitives = json.load(fp)
 
     # Generate primitives
     primitive_dir = (working_dir / '2_primitives')
     if '2_primitives' in steps_to_run:
         primitive_dir.mkdir(exist_ok=True)
         for block_name, block_args in primitives.items():
-            logger.debug(f"Generating primitive: {block_name}")
-            uc = generate_primitive(block_name, **block_args, pdkdir=pdk_dir, outputdir=primitive_dir, netlistdir=netlist_dir)
+            logger.info(f"Generating primitive: {block_name} {block_args}")
+            if block_args['primitive'] != 'generic':
+                primitive_def = ckt_data.find(block_args['primitive'])
```

Comment:
This will be difficult now as primitive generators need the circuit definitions. One way I can think of is to pickle dump the database and read it again. Any other suggestions?

Reply:
fixed.